### PR TITLE
JEP 360 sealed classes instrumentation restrictions

### PR DESCRIPTION
**Pull request by theresa-m**
Related to: https://github.com/eclipse/openj9/issues/9626

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

**Comment by theresa-m** [comment]
Copyright failure refers to new MANIFEST.mf file which I don't believe can have comments.

File: runtime/util/hshelp.c [hunk 1]
```diff
@@ -3220,6 +3220,54 @@ verifyRecordAttributesAreSame(J9ROMClass *originalROMClass, J9ROMClass *replacem
...
+verifyPermittedSubclassAttributeContentsMatch(J9ROMClass *originalROMClass, J9ROMClass *replacementROMClass)
+{
+	jvmtiError rc = JVMTI_ERROR_NONE;
+
+	if (J9ROMCLASS_IS_SEALED(originalROMClass) && J9ROMCLASS_IS_SEALED(replacementROMClass)) {
+		U_32* originalPermittedSubclassesCountPtr = getNumberOfPermittedSubclassesPtr(originalROMClass);
+		U_32* replacementPermittedSubclassesCountPtr = getNumberOfPermittedSubclassesPtr(replacementROMClass);
+
+		if (*originalPermittedSubclassesCountPtr == *replacementPermittedSubclassesCountPtr) {
+			U_32 i, j;
```
Comment (gacholio) [inline runtime/util/hshelp.c:3234]: Please declare each variable on its own line and initialize them.

Comment (gacholio) [reply runtime/util/hshelp.c:3234]: Also, `j` should be declared inside the loop.

File: runtime/util/hshelp.c [hunk 2]
```diff
@@ -3220,6 +3220,54 @@ verifyRecordAttributesAreSame(J9ROMClass *originalROMClass, J9ROMClass *replacem
...
+					J9UTF8* replacementSubclassNameUTF = permittedSubclassesNameAtIndex(replacementPermittedSubclassesCountPtr, adjustedIndex);
+
+					if (J9UTF8_EQUALS(originalSubclassNameUTF, replacementSubclassNameUTF)) {
+						foundMatchingSubclass = TRUE;
+						break;
+					}
+				}
+
+				/* check if matching subclass was found. */
+				if (FALSE == foundMatchingSubclass) {
```
Comment (gacholio) [inline runtime/util/hshelp.c:3254]: Use `!` for booleans.

Comment (theresa-m) [reply runtime/util/hshelp.c]: updated.

**Comment by gacholio** [comment]
jenkins test sanity,extended win jdk15

**Pull request merged**